nexus/db-queries/src/db/datastore/dataset.rs

```diff
@@ -180,6 +182,56 @@ impl DataStore {
 
         Ok(all_datasets)
     }
+
+    pub async fn dataset_on_in_service_physical_disk(
```

Comment:
`on_in` reads a little wonky to me. I thought it was a typo at first. Maybe something like `in_service_physical_disk_contains_dataset` would work?

Reply:
yeah, this sucks. changed to `dataset_physical_disk_in_service` in 9e8fcc147

Reply:
Thanks.